lib/protocol/rest_xml.js

```diff
@@ -43,7 +43,8 @@ function extractError(resp) {
   if (data.Code) {
     resp.error = util.error(new Error(), {
       code: data.Code,
-      message: data.Message
+      message: data.Message,
+      region: data.Region
```

Comment:
There shouldn't be any changes needed in this file since the `extractError` in lib/services/s3.js will handle setting the region.  This functionality is specific to S3 so would like to keep the logic contained there.